crates/pipeline_manager/src/main.rs

```diff
@@ -1348,3 +1419,219 @@ async fn pipeline_delete(state: WebData<ServerState>, req: HttpRequest) -> impl
+    state
+        .db
+        .lock()
+        .await
+        .list_connectors()
+        .await
+        .map(|connectors| {
+            HttpResponse::Ok()
+                .insert_header(CacheControl(vec![CacheDirective::NoCache]))
+                .json(connectors)
+        })
+        .unwrap_or_else(|e| http_resp_from_error(&e))
+}
+
+/// Request to create a new connector.
+#[derive(Deserialize, ToSchema)]
+pub(self) struct NewConnectorRequest {
+    /// connector name.
+    name: String,
+    /// connector description.
+    description: String,
+    /// connector type.
+    typ: ConnectorType,
+    /// connector config.
+    config: String,
+}
+
+/// Response to a connector creation request.
+#[derive(Serialize, ToSchema)]
+struct NewConnectorResponse {
+    /// Unique id assigned to the new connector.
+    connector_id: ConnectorId,
+}
+
+/// Create a new connector configuration.
+#[utoipa::path(
+    request_body = NewConnectorRequest,
+    responses(
+        (status = OK, description = "connector successfully created.", body = NewConnectorResponse),
+    ),
+    tag = "Connector"
+)]
+#[post("/connector")]
+async fn new_connector(
+    state: WebData<ServerState>,
+    request: web::Json<NewConnectorRequest>,
+) -> impl Responder {
+    state
+        .db
+        .lock()
+        .await
+        .new_connector(
+            &request.name,
+            &request.description,
+            request.typ,
+            &request.config,
+        )
+        .await
+        .map(|connector_id| {
+            HttpResponse::Ok()
+                .insert_header(CacheControl(vec![CacheDirective::NoCache]))
+                .json(&NewConnectorResponse { connector_id })
+        })
+        .unwrap_or_else(|e| http_resp_from_error(&e))
+}
+
+/// Request to update an existing data-connector.
+#[derive(Deserialize, ToSchema)]
+struct UpdateConnectorRequest {
+    /// connector id.
+    connector_id: ConnectorId,
+    /// New connector name.
+    name: String,
+    /// New connector description.
+    description: String,
+    /// New config YAML. If absent, existing YAML will be kept unmodified.
+    config: Option<String>,
+}
+
+/// Response to a config update request.
+#[derive(Serialize, ToSchema)]
+struct UpdateConnectorResponse {}
+
+/// Update existing project connector.
+///
+/// Updates project config name and, optionally, code.
+/// On success, increments config version by 1.
+#[utoipa::path(
+    request_body = UpdateConnectorRequest,
+    responses(
+        (status = OK, description = "connector successfully updated.", body = UpdateConnectorResponse),
+        (status = NOT_FOUND
+            , description = "Specified `connector_id` does not exist in the database."
+            , body = ErrorResponse
+            , example = json!(ErrorResponse::new("Unknown connector id '5'"))),
+    ),
+    tag = "Connector"
+)]
+#[patch("/connectors")]
+async fn update_connector(
```

Comment:
@ryzhyk can confirm, but I think you'd also need to update the python API + tests for these changes I think.

Reply:
That's correct.  I can help with that.  We should probably mark this as draft PR for now.